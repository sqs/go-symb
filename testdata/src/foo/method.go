package foo

type R int

func (r R) M() {}

func init() {
	R(7).M()
}
