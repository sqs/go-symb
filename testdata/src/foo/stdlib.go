package foo

import (
	"flag"
	"fmt"
)

func main() {
	flag.Usage = func() {}
	flag.Parse()
	fmt.Println("Hello!")
}
