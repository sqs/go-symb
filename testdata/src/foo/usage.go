package foo

func B() {
	eB, fB, _ := A("a", "b", true)
	eB = fB
	fB = eB
}
