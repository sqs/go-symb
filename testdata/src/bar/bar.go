package bar

import "foo"

func main() {
	foo.A("b", "c", true)
}
