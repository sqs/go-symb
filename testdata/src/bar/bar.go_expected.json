[
  {
    "Expr": "bar",
    "Ident": "bar",
    "IdentPos": {
      "Filename": "testdata/src/bar/bar.go",
      "Offset": 8,
      "Line": 1,
      "Column": 9
    },
    "ExprType": "",
    "Pkg": {
      "Isa": "Package",
      "Name": "bar",
      "ImportPath": "bar"
    },
    "FileName": "bar",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "Package",
      "Name": "bar",
      "ImportPath": "bar"
    },
    "Local": false,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "main",
    "Ident": "main",
    "IdentPos": {
      "Filename": "testdata/src/bar/bar.go",
      "Offset": 32,
      "Line": 5,
      "Column": 6
    },
    "ExprType": "func()",
    "Pkg": {
      "Isa": "Package",
      "Name": "bar",
      "ImportPath": "bar"
    },
    "FileName": "bar",
    "ReferPos": {
      "Filename": "testdata/src/bar/bar.go",
      "Offset": 32,
      "Line": 5,
      "Column": 6
    },
    "ReferObj": {
      "Isa": "Func",
      "Pkg": {
        "Isa": "Package",
        "Name": "bar",
        "ImportPath": "bar"
      },
      "Name": "main",
      "Type": "func()"
    },
    "Local": false,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "foo",
    "Ident": "foo",
    "IdentPos": {
      "Filename": "testdata/src/bar/bar.go",
      "Offset": 42,
      "Line": 6,
      "Column": 2
    },
    "ExprType": "",
    "Pkg": {
      "Isa": "Package",
      "Name": "bar",
      "ImportPath": "bar"
    },
    "FileName": "bar",
    "ReferPos": {
      "Filename": "testdata/src/bar/bar.go",
      "Offset": 20,
      "Line": 3,
      "Column": 8
    },
    "ReferObj": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": "foo"
    },
    "Local": false,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "foo.A",
    "Ident": "A",
    "IdentPos": {
      "Filename": "testdata/src/bar/bar.go",
      "Offset": 46,
      "Line": 6,
      "Column": 6
    },
    "ExprType": "func(b·4 string, c·5 string, d·6 bool) (e·1 int, f·2 int, g·3 uint)",
    "Pkg": {
      "Isa": "Package",
      "Name": "bar",
      "ImportPath": "bar"
    },
    "FileName": "bar",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "Func",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": "foo"
      },
      "Name": "A",
      "Type": "func(b·4 string, c·5 string, d·6 bool) (e·1 int, f·2 int, g·3 uint)"
    },
    "Local": false,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "true",
    "Ident": "true",
    "IdentPos": {
      "Filename": "testdata/src/bar/bar.go",
      "Offset": 58,
      "Line": 6,
      "Column": 18
    },
    "ExprType": "bool",
    "Pkg": {
      "Isa": "Package",
      "Name": "bar",
      "ImportPath": "bar"
    },
    "FileName": "bar",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "Const",
      "Pkg": null,
      "Name": "true",
      "Type": "untyped boolean",
      "Val": true
    },
    "Local": false,
    "Universe": true,
    "IsDecl": false
  }
]
